daemon/cluster/cluster.go

```diff
@@ -74,28 +76,52 @@ var defaultSpec = types.Spec{
 }
 
 type state struct {
+	// LocalAddr is this machine's local IP or hostname, if specified.
+	LocalAddr string
+	// RemoteAddr is the address that was given to "swarm join. It is used
+	// to find LocalAddr if necessary.
+	RemoteAddr string
+	// ListenAddr is the address we bind to, including a port.
 	ListenAddr string
+	// AdvertiseAddr is the address other nodes should connect to,
+	// including a port.
+	AdvertiseAddr string
+}
+
+// NetControllerProvider exposes a function for retrieving the network
+// controller.
+type NetControllerProvider interface {
+	NetController() libnetwork.NetworkController
```

Comment:
I could make a function that returns the subnets of networks that Docker manages. Do you think that would be better?